.github/workflows/pre-commit.yml

```diff
@@ -45,7 +47,9 @@ jobs:
       - name: Checkout source code
         uses: actions/checkout@v3
         with:
-          fetch-depth: 2
+          fetch-depth: 0
+      - name: Fetch all branches and tags
+        run: git fetch --all
```

Comment:
Do not think this is needed, the fetch-depth or any history to build for the e2e tests only.